extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultSystemBackendEngine.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.vault;
+
+import io.quarkus.vault.runtime.config.VaultRuntimeConfig;
+import io.quarkus.vault.sys.VaultHealth;
+import io.quarkus.vault.sys.VaultHealthStatus;
+import io.quarkus.vault.sys.VaultInit;
+import io.quarkus.vault.sys.VaultSealStatus;
+
+/**
+ * This service provides access to the system backend.
+ *
+ * @see VaultRuntimeConfig
+ */
+public interface VaultSystemBackendEngine {
+
+    /**
+     * Initializes a new Vault.
+     * 
+     * @param secretShares specifies the number of shares to split the master key into.
+     * @param secretThreshold specifies the number of shares required to reconstruct the master key.
+     * @return Vault Init.
+     */
+    VaultInit init(int secretShares, int secretThreshold);
+
+    /**
+     * Check the health status of Vault.
+     * Returns Vault health status code only by using HTTP HEAD requests.
+     * It is faster than calling {@link #healthStatus() healthStatus()} method which uses HTTP GET to return a complete
+     * VaultHealthStatus state.
+     *
+     * @return Vault Health Status.
+     */
+    VaultHealth health();
```

Comment:
@lordofthejars So do you think that `VaultHealth` is more appropriate than `VaultHealthStatus` given that is only has a status code ? Wouldn;t `VaultHealthStatusCode` be more appropriate ?